Windows quick start (#538)
jellyfin/jellyfin.org: docs/general/quick-start/windows.md

    * Click `Open PowerShell window here`
  * Windows 11
    * Right click on blank space
    * Click `Open in Windows Terminal` or `Open in Terminal`

## Jellyfin Install

### Install

1. Download installer exe and sha256sum from [the official repo](https://repo.jellyfin.org/releases/server/windows/stable/). The files to download are `installer/jellyfin_x.y.z_windows-x64.exe` and `installer/jellyfin_x.y.z_windows-x64.exe`.

    ![Windows Download Page](/images/docs/quick-start/windows/jellyfin-0.png)

2. Once the download finishes, open the sha256sum in a text editor
3. Open a Terminal in your Downloads folder and enter this command: `certutil -hashfile .\jellyfin_x.y.z_windows-x64.exe SHA256`, replace `jellyfin_x.y.z_windows-x64.exe` with the .exe file you just downloaded.
4. Wait for the command to finish and see if it matches the value in the sha256sum file. If the values don't match, at least one of the files is corrupted and both files should be redownloaded.

    ![Checksum Match](/images/docs/quick-start/windows/jellyfin-1.png)

    Values match, files are good

    ![Checksum Doesn't Match](/images/docs/quick-start/windows/jellyfin-2.png)

>     Values don't match, possibly corrupted file(s)

Reviewer: I don't think this is worth mentioning, 99.999% of the downloads succeed and won't end up in corrupt files.

> 5. Run and follow the installer

Reviewer: This part might need some more information. Which steps does the installer have, what options do I select (e.g. don't install as a service because that breaks transcoding)

6. Look for a Jellyfin icon in your system tray. If it isn't present, search for `Jellyfin Tray App` and run it. No windows will popup after running this application.

    ![Jellyfin Tray App Search](/images/docs/quick-start/windows/jellyfin-3.png)

    Jellyfin Tray App

7. Allow Firewall Access when Prompted

    ![Windows Firewall Prompt](/images/docs/quick-start/windows/jellyfin-4.png)

    Windows Firewall Prompt

> 8. Open a browser and go to [http://localhost:8096](http://localhost:8096) to finish the setup process. Leave the `Enable Automatic Port Mapping` option unchecked.

Reviewer: Why do I need to leave the option unchecked?

> ### Hardware Accel

Reviewer: ```suggestion
### Hardware Acceleration
```

> See [Hardware Acceleration](/docs/general/administration/hardware-acceleration/)

Reviewer: ```suggestion
Read the dedicated [Hardware Acceleration](/docs/general/administration/hardware-acceleration/) page to configure hardware acceleration on you install.
```

7. Most Routers will show a WAN IP in the index. Check if the WAN IP matches the IP you noted down in step 5. If it doesn't, you do not have a routable IP address. In this case, you should stop following this guide and look for other options for remote access.

    ![ASUS router settings page](/images/docs/quick-start/windows/router-4.png)

    ASUS Router Settings Page

8. Assign your Jellyfin host a static local IP. The option is usually under `LAN` and named `DHCP Server` or `DHCP`. If you assign a different IP than what it currently has, you may need to reconnect the internet of this PC for settings to apply properly. In this case, I am assigning 192.168.50.142 to my PC.

    ![Static DHCP](/images/docs/quick-start/windows/router-5.png)

9. Find `Port Forwarding` settings in your router. This setting is usually under `WAN`, `Firewall`, `Security` or `NAT`and usually named `Port Forwarding` or `Virtual Server`. `Port Trigger` is NOT the setting you are looking for.

10. Forward ports 80 and 443 to the IP you assigned previously

    ![Port Forwarding](/images/docs/quick-start/windows/router-6.png)

> ### Install Caddy

Reviewer: Might be worth adding a small text explaining what a reverse proxy is and listing some alternatives to Caddy.

1. Go to the [official Caddy server download page](https://caddyserver.com/download)
2. Choose `Windows amd64` for the platform and click download. Leave the standard features checkbox checked. You do NOT need any extra features.

    ![Caddy Download](/images/docs/quick-start/windows/caddyinst-1.png)

> 3. Create a new folder under `C:\` and name it `caddy`

Reviewer: I would highly suggest not adding a folder to your root. Something like `C:\Program Files\Caddy` would make more sense.

4. Attempt to access `http://<your IP here>` using another internet connection on another device such as your mobile data. For this case, the IP has changed to `1.171.xxx.xxx` so I will be accessing `http://1.171.xxx.xxx` using my mobile data on my phone.

5. The page should look like the picture below. If it doesn't look this way or fails to connect, your ISP may be blocking incoming http connections or your IP is not routable. In this case, you should stop following this guide and look for other remote access options.

    ![test success ios](/images/docs/quick-start/windows/httptest-3.png)

    Screenshot taken on iOS using Safari

6. Press `Ctrl + C` in the terminal to exit Caddy, then close the terminal.

### DDNS

1. Login to DuckDNS on the top right at the [DuckDNS homepage](https://www.duckdns.org/)
2. Add a new domain in the middle. I am using jellyfintest12345.duckdns.org for this example but you can use anything as long as it hasn't been taken

    ![duckdns dashboard](/images/docs/quick-start/windows/ddns-1.png)

3. Download the latest Windows C# client zip from [its Github releases page](https://github.com/XWolfOverride/DuckDNS/releases)

> 4. Create a new subfolder under `C:\` and name it `duckdns`

Reviewer: Same as the Caddy comment, don't add it to your root

8. Make sure that no errors appeared and there is a yellow duck in your system tray. If the duck is red, check your configuration.

9. hit `Win + R`, enter `taskschd.msc` and click `OK`

    ![run task scheduler](/images/docs/quick-start/windows/ddns-5.png)

10. Click on `Create basic task` on the right panel.
11. Enter a descriptive Name and a description, one that you can identify in the future and click next
12. Choose `When I log on` for the Task Trigger and click Next
13. Choose `Start a program` for the Action
14. Enter `C:\duckdns\DuckDNS.exe` in the Program/script field

    ![create duckdns task](/images/docs/quick-start/windows/ddns-6.png)

15. Check `Open the Properties dialog` checkbox and click `Finish`

    ![finish wizard](/images/docs/quick-start/windows/ddns-7.png)

16. Go to the `Settings` Pane in the dialog and uncheck the `Stop the task if it runs longer than` checkbox, then click `OK`

    ![duckdns task properties](/images/docs/quick-start/windows/ddns-8.png)

### Reverse Proxy with Caddy

1. Go to your caddy folder (the one you created in the Install Caddy section which should be `C:\caddy`)

> 2. create a text file named `CaddyFile` (Remove the .txt suffix). You will be asked to confirm if you want to change the file extension, click `Yes`.

Reviewer: Should be `Caddyfile`, not `CaddyFile`

* Show file extensions in Explorer ([Guide](https://www.howtogeek.com/205086/beginner-how-to-make-windows-show-file-extensions/))
* Setup auto login for your user account if you haven't already ([Guide](https://www.lifewire.com/how-do-i-auto-login-to-windows-2626066))
* Make your media files accessible on this computer by this user
* Install latest updates ([Guide](https://support.microsoft.com/en-us/windows/get-the-latest-windows-update-7d20e88c-0568-483a-37bc-c3885390d212))
* Setup remote access (optional). Popular options include [Teamviewer](https://www.teamviewer.com/), [AnyDesk](https://anydesk.com/), [RustDesk](https://rustdesk.com/), [Google Remote Desktop](https://remotedesktop.google.com/) and [Parsec](https://parsec.app/)

## Notes

* How to open a Terminal from current location in Explorer
  * Windows 10
    * Right click on blank space while holding down `Shift`
    * Click `Open PowerShell window here`
  * Windows 11
    * Right click on blank space
    * Click `Open in Windows Terminal` or `Open in Terminal`

> ## Jellyfin Install

Reviewer 2: ```suggestion
## Install Jellyfin
```

Either this or switch the order on the Caddy heading

    ![Create Caddy folder](/images/docs/quick-start/windows/caddyinst-2.png)

4. move the downloaded exe file to the folder and rename it to `caddy.exe`. Confirm the operation if prompted.

    ![Place Caddy exe](/images/docs/quick-start/windows/caddyinst-3.png)

5. go to `This PC` in Explorer, right click on blank space and click on `properties`

    ![This PC Right Click Menu](/images/docs/quick-start/windows/caddyinst-4.png)

6. Click `Advanced System Settings`. it will be after `Device Specifications` on Windows 11, or in the left sidebar on Windows 10
7. Click on `Environment Variables` in the popup and doubleclick on `Path` under the `User variables for user` in the second popup

    ![Environment Variables](/images/docs/quick-start/windows/caddyinst-5.png)

8. Click on `New` in the popup and enter `C:\Program Files\caddy`, then click `OK` on the bottom right. DO NOT change the other entries as messing with them can result in broken programs and/or a broken system.

    ![New PATH value](/images/docs/quick-start/windows/caddyinst-6.png)

9. Click on `OK` for the other 2 popups to save and apply settings
10. Reboot to make sure settings apply poperly

> ### Test if http is routable

Reviewer 2: ```suggestion
### Test if HTTP is routable
```

    * if you want your jellyfin instance behind the bare domain (subdomain.example.com / jellyfin.example.com):

        ```Caddyfile
        <your (sub)domain here>
        
        reverse_proxy 127.0.0.1:8096
        ```

    * if you want your jellyfin instance under a subpath (example.com/jellyfin):

        ```Caddyfile
        <your (sub)domain here>

        redir /jellyfin /jellyfin/
        reverse_proxy /jellyfin/* 127.0.0.1:8096
        ```

3. Save the file and close the text editor

4. Copy this file to your caddy folder (the one you created in the Install Caddy section which should be `C:\Program Files\caddy`). Confirm the operation if prompted.

5. If you have chosen to use a subpath for Jellyfin, go to your Jellyfin dashboard (open [http://localhost:8096](http://localhost:8096) on your server, click the 3 bars on the top right then select `Dashboard`) and go to the `Networking` menu on the side bar. Then enter the subpath you have chosen to use and save. Then restart the Jellyfin server (Stop then start the server from the system tray icon)

    ![Networking Settings](/images/docs/quick-start/windows/reverseproxy-2.png)

> 6. open a terminal in the caddy folder and run `caddy run`. If there are errors, check your configs and try again. Keep this window running.

Reviewer 2: ```suggestion
6. Open a terminal in the caddy folder and run `caddy run`. If there are errors, check your configs and try again. Keep this window running.
```

> 7. go to `http://<your (sub)domain here>` or `http://<your (sub)domain here>/jellyfin` in a browser to test if it is working. You should be able to see the Jellyfin Login Page.

Reviewer 2: ```suggestion
7. Go to `http://<your (sub)domain here>` or `http://<your (sub)domain here>/jellyfin` in a browser to test if it is working. You should be able to see the Jellyfin Login Page.
```
